core/src/main/java/org/mule/interceptor/AbstractEnvelopeInterceptor.java

```diff
@@ -59,6 +65,69 @@ public MuleEvent process(MuleEvent event) throws MuleException
         return resultEvent;
     }
 
+    @Override
+    protected MuleEvent processNonBlocking(final MuleEvent event) throws MuleException
+    {
+        final long startTime = System.currentTimeMillis();
+        final ProcessingTime time = event.getProcessingTime();
+        MuleEvent responseEvent = event;
+        boolean exceptionWasThrown = true;
```

Comment:
This is from blocking implementation thats not correctly implemented in this case. Will update.